woocommerce_fusion/tasks/sync_sales_orders.py

```diff
@@ -630,7 +630,7 @@ def create_missing_items(self, items_list, woocommerce_site):
 				row.woocommerce_server = woocommerce_site
 				item.flags.ignore_mandatory = True
 				if item.item_defaults:
-					item.item_defaults.default_supplier = self.settings.default_supplier
+					item.item_defaults[0]['default_supplier'] = self.settings.default_supplier
```

Comment:
What if `item_defaults[0].company` is different from `settings.company`? What you need to do is to go through the objects in `item_defaults` and pick the one where `company` matches `settings.company`. If there's none that matches, add a new one else update the one that matches